### ISSUE TITLE
TypeError('"string" must be a string, Buffer, or ArrayBuffer') when logging a string

### ISSUE THREAD

**Issue by mrister** [loggly/node-loggly-bulk#10]
So When I do the example like in the readme (With proper Loggly credentials).
https://github.com/loggly/node-loggly-bulk#getting-started

And If i do a `client.log('test')` I get:
`TypeError('"string" must be a string, Buffer, or ArrayBuffer');`

I am not sure the tests, in general here, are passing either.

**Comment by mostlyjason** [comment]
@Shwetajain148 can you reproduce this on your end?

**Comment by Shwetajain148** [comment]
@mrister I am not able to reproduce the issue. Can you please share your .js file that is creating the issue for you. Also, share the version of node and details of the environment in which you are setting up logging and observing this issue.

cc: @mostlyjason

**Comment by mrister** [comment]
@mostlyjason Node: v7.7.2
Here is the code:
```js
// https://github.com/loggly/node-loggly-bulk#getting-started
// did: npm i node-loggly-bulk
const loggly = require('node-loggly-bulk');

const client = loggly.createClient({
  token: '<My TOKEN>',
  subdomain: '<My SUBDOMAIN>',
  json:true
});

client.log('test');
```

**Comment by mrister** [comment]
Works on node 6.9.1

**Comment by mostlyjason** [comment]
Closing since this issue is fixed. If it reoccurs please reopen.

**Comment by jingram-classy** [comment]
Seeing this again with Node v8.8.0:
TypeError: "string" must be a string, Buffer, or ArrayBuffer
    at Function.byteLength (buffer.js:479:11)
    at exports.Loggly.Loggly.log (/classy/pay/node_modules/node-loggly-bulk/lib/loggly/client.js:161:32)
    at Bunyan2Loggly._processBuffer (/classy/pay/node_modules/bunyan-loggly/index.js:48:32)
    at Bunyan2Loggly._checkBuffer (/classy/pay/node_modules/bunyan-loggly/index.js:61:21)
    at Bunyan2Loggly.write (/classy/pay/node_modules/bunyan-loggly/index.js:37:10)
    at Logger._emit (/classy/pay/node_modules/bunyan/lib/bunyan.js:923:22)
    at Logger.debug (/classy/pay/node_modules/bunyan/lib/bunyan.js:1045:24)
    at /classy/pay/bin/www:19:7
    at /classy/pay/config.js:294:5
    at __NR_wrappedThenHandler (/classy/pay/node_modules/newrelic/lib/instrumentation/promise.js:468:25)
    at runCallback (timers.js:785:20)
    at tryOnImmediate (timers.js:747:5)
    at processImmediate [as _immediateCallback] (timers.js:718:5)
From previous event:
    at Promise.__NR_wrappedThen (/classy/pay/node_modules/newrelic/lib/instrumentation/promise.js:426:23)
    at Object.config.init (/classy/pay/config.js:291:22)
    at Object.<anonymous> (/classy/pay/bin/www:12:8)
    at Module._compile (module.js:612:30)
    at Object.Module._extensions..js (module.js:623:10)
    at Module.load (module.js:531:32)
    at tryModuleLoad (module.js:494:12)
    at Function.Module._load (module.js:486:3)
    at Function.Module.runMain (module.js:653:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
root@ip-10-21-201-241:/var/log/upstart# node -v
v8.8.0

**Comment by mostlyjason** [comment]
Reopening since it's been reported here as well https://github.com/MauriceButler/bunyan-loggly/issues/23. FYI bunyan-loggly is a community supported library so we have not tested that one yet. The fix we made earlier for winston might affect bunyan differently.

**Comment by jingram-classy** [comment]
NOTE: Not reproducible on Node 8.4 but always reproducible on Node 8.8.

**Comment by jingram-classy** [comment]
FYI, for bunyan-loggly users, I was able to for bunyan-loggly and downgrade node-loggly-balk to 2.0.2 as a workaround.  Works fine for now.

**Comment by mostlyjason** [comment]
Addressed in 2.1.0 please reopen if it keeps happening after updating.

**Comment by mostlyjason** [comment]
Hey guys we temporarily had to roll back this change in 2.2.0 so this may pop up again, but we are working on a fix which should be ready in a few days.

**Comment by mrister** [comment]
@mostlyjason I suggest you reopen to not forget this issue and close once confirmed what exact version when it will be fixed.

**Comment by mostlyjason** [comment]
Thanks I'm reopening it now

**Comment by tinder-dyakobian** [comment]
I am experiencing this issue in latest update

**Comment by Shwetajain148** [comment]
@mostlyjason I have published a beta package and left a comment including the instruction to install the beta package on the issue https://github.com/MauriceButler/bunyan-loggly/issues/23#issuecomment-347842721.

**Comment by mostlyjason** [comment]
This should be fixed in version 2.2.1. Closing for now, but please reopen if the issue reoccurs after updating.
